### PR TITLE
SCANNPM-2 Use new endpoints & checksum/download logic & Do not use axios instance when fetching absolute URLs

### PR DESCRIPTION
**Pull request by 7PH**
[ticket](https://sonarsource.atlassian.net/browse/SCANNPM-25)

### CODE REVIEW

File: src/request.ts [hunk 1]
```diff
@@ -95,31 +101,45 @@ export async function getHttpAgents(
   return agents;
 }
 
+export function resetAxios() {
+  _axiosInstances = null;
+}
```
Comment (7PH) [inline src/request.ts:106]: That is only needed for the test environment to reset the internal state of this module

File: src/platform.ts [hunk 2]
```diff
@@ -26,15 +26,15 @@ export function getArch(): NodeJS.Architecture {
...
 
 export function isWindows() {
-  return process.platform === 'win32';
+  return process.platform.startsWith('win');
 }
 
 export function isMac() {
-  return process.platform === 'darwin';
+  return process.platform.startsWith('darwin');
 }
```
[the remark this answers is not in the document]
Comment (vdiez) [reply src/platform.ts:38]: The range of possible values is very well specified, that's why I preferred a more explicit version

**Review by vdiez — APPROVED** [review]
there are many `'https://api.sonarcloud.io'` that could be replaced with `SONARCLOUD_API_BASE_URL`. And same with `'https://sonarcloud.io'`. Otherwise looks good!

File: src/platform.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (7PH) [reply src/platform.ts:38]: Makes sense. I recall a discussion where users could potentially build their own NodeJS and specify a custom value for `process.platform`, but that's probably a very far stretch, I'm reverting this change

**Comment by 7PH** [comment]
SC issues are in the main branch, I'm merging this

**Pull request merged**